File: BackEnd/src/CollectFlow.Domain/Entities/Invoice.cs
using CollectFlow.Domain.Enums;

namespace CollectFlow.Domain.Entities;

public class Invoice : BaseEntity
{
    public Guid TenantId { get; set; }
    public Tenant Tenant { get; set; } = default!;

    public Guid CustomerId { get; set; }
    public Customer Customer { get; set; } = default!;

    public string InvoiceNumber { get; set; } = string.Empty;
    public DateOnly IssueDate { get; set; }
    public DateOnly DueDate { get; set; }
    public decimal Amount { get; set; }
    public decimal Balance { get; set; }
    public string Currency { get; set; } = "USD";
    public InvoiceStatus Status { get; set; } = InvoiceStatus.Sent;
}

File: FrontEnd/src/types/invoice.ts
export type InvoiceStatus = 'Draft' | 'Sent' | 'PartiallyPaid' | 'Paid' | 'Overdue' | 'WrittenOff' | number;

export type Invoice = {
  id: string;
  tenantId: string;
  customerId: string;
  customerName: string;
  customerCompanyName: string;
  invoiceNumber: string;
  issueDate: string;
  dueDate: string;
  amount: number;
  balance: number;
  currency: string;
  status: InvoiceStatus;
  daysOverdue: number;
  createdAtUtc: string;
};

export type CreateInvoiceRequest = {
  tenantId: string;
  customerId: string;
  invoiceNumber: string;
  issueDate: string;
  dueDate: string;
  amount: number;
  balance: number;
  currency: string; 
};
